x/likenft/keeper/msg_server_update_class.go

```diff
@@ -82,18 +82,17 @@ func (k msgServer) UpdateClass(goCtx context.Context, msg *types.MsgUpdateClass)
 		return nil, types.ErrFailedToUpdateClass.Wrapf("%s", err.Error())
 	}
 
-	if !originalConfig.IsBlindBox() && updatedConfig.IsBlindBox() {
-		// Enqueue class if enabled after update
+	// Dequeue original reveal schedule
+	if originalConfig.IsBlindBox() {
+		k.RemoveClassRevealQueueEntry(ctx, originalConfig.BlindBoxConfig.RevealTime, class.Id)
+	}
+
+	// Enqueue new reveal schedule
+	if updatedConfig.IsBlindBox() {
 		k.SetClassRevealQueueEntry(ctx, types.ClassRevealQueueEntry{
 			ClassId:    class.Id,
 			RevealTime: updatedConfig.BlindBoxConfig.RevealTime,
 		})
```

Comment:
so the update function is not used anymore? but i guess we can keep it just in case